src/Library/demos/Welcome/main.py

```diff
@@ -0,0 +1,30 @@
+import gi
+
+gi.require_version("Gtk", "4.0")
+gi.require_version("Adw", "1")
+from gi.repository import Gtk, Adw
+import workbench
+
+
+def greet(_widget):
+    dialog = Adw.MessageDialog(body="Hello World!", transient_for=workbench.window)
+
+    dialog.add_response("ok", "Ok")
+    dialog.connect("response", handle_response)
+    dialog.present()
+
+
+def handle_response(dialog: Adw.MessageDialog, response: str):
+    print(response)
+    dialog.close()
+
+
+subtitle_box: Gtk.Box = workbench.builder.get_object("subtitle")
```

Comment:
I wonder how useful these type annotations are.
Even if Workbench would get mypy support at some point: does PyGObject even expose type information?

Reply:
Yes, via https://github.com/pygobject/pygobject-stubs/pull/153

I wanted to leave them in the demo where useful. 

This isn't really for strict type checkers. For this to pass mypy or pyright, a typing.cast would need to be done here.

Reply:
> Yes, via https://github.com/pygobject/pygobject-stubs/

Interesting, thanks for the link

> I wanted to leave them in the demo where useful.

I actually agree that they make the code more readable :)
However, I also think that unchecked type hints do more harm than good in the end. Demos get adapted from time to time, and type hints are easily missed. How do you feel about this @sonnyp?

Reply:
There is a difference though. Unlike Vala and Rust, Python type hints can be incorrect, but the program would still work fine

Reply:
Seeing as there's already some tooling for Rust for running rustfmt, we COULD do something similiar here, like running mypy or pyright (the latter is more strict but faster). And refuse to run if it fails.
Since for the other external languages we also need to compile, the time it would take to wait would probably still be faster than that.